smartcontracts/contracts/W3bstreamProver.sol

```diff
@@ -6,15 +6,16 @@ import {ERC721Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC72
 
 contract W3bstreamProver is OwnableUpgradeable, ERC721Upgradeable {
     event OperatorSet(uint256 indexed id, address indexed operator);
-    event NodeTypeUpdated(uint256 indexed id, uint256 typ);
+    event NodeTypeAdded(uint256 indexed id, uint256 typ);
+    event NodeTypeDeleted(uint256 indexed id, uint256 typ);
     event ProverPaused(uint256 indexed id);
     event ProverResumed(uint256 indexed id);
     event MinterSet(address minter);
 
     address public minter;
     uint256 nextProverId;
 
-    mapping(uint256 => uint256) _nodeTypes;
+    mapping(uint256 => mapping(uint256 => bool)) _nodeTypes;
```

Comment:
can still use `mapping(uint256 => uint256)` as node types, because node type NFT id is start 1, you can use 0 to check that doesn't set node type.